spec/tictactoe_spec.rb

```diff
@@ -7,7 +7,25 @@
   let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
 
   it "runs the game loop " do
-    allow(test_ttt.game_loop).to receive(:game_loop) { }
-    test_ttt.game_loop
+    expect(test_ttt.game_loop).to receive(:game_loop).once
+    expect(game_io).to receive(:puts_message)
+    expect(game_io).to receive(:get_input).and_return("N")
+    test_ttt.play_game
+  end
+
+  it "plays the game twice " do
+    expect(test_ttt.game_loop).to receive(:game_loop).twice
```

Comment:
A little awkward to stub a method (or variable) of the thing you're testing.

Don't worry about this for now, but just keep that in mind for the future - it's sometimes a sign that you're doing something funky somewhere.